scripts/west_commands/sign.py

```diff
@@ -520,6 +520,81 @@ def mcuboot_swap_validate(mcbdtconf, bin_size, quiet=False):
 
         return (write_alignment, cp_addr, cp_size)
 
+    @staticmethod
+    def mcuboot_swap_scratch_validate(mcbdtconf, bin_size, quiet=False):
+        '''Check whether DT configuration of partitions is valid for MCUboot swap-scratch algorithm.
+
+        If configuration is valid, returns parameters needed by imgtool sign command.
+
+        :param mcbdtconf: MCUboot configuration reader
+        :type: MCUbootDTConfig
+        :param bin_size: size of application binary; this is size application will take
+                         on flash.
+        :type bin_size: int
+        :param quiet: suppress extra diagnostic information
+        :type quiet: Bool
+        :returns: (flash alignment, application image offset on flash, image size)
+        :rtype: (int, int, int)
+        '''
+
+        code = mcbdtconf.get_code_partition()
+
+        if 'slot0_partition' not in code.labels:
+            log.die("west sign, for MCUboot in swap-scratch mode, expects DT chosen"
+                    " zephyr,code-partition to point to node labeled"
+                    " slot0_partition. Currently selected node is labeled"
+                    " " + str(code.labels))
+
+        if 'slot1_partition' and 'scratch_partition' in code.labels:
+            log.die("chosen zephyr,code-partition may not be slot0_partition,"
+                    " slot1_partition and scratch_partition at the same time")
```

Comment:
This cannot happen

Reply:
Unless somebody does two labels to the same node. I can remove these checks, if not wanted.